manifests/init.pp

```diff
@@ -750,41 +802,57 @@
     validate_array($sshd_config_allowgroups_real)
   }
 
-  package { $packages_real:
-    ensure    => installed,
-    source    => $ssh_package_source_real,
-    adminfile => $ssh_package_adminfile_real,
+  if $packages_real != undef {
+    package { $packages_real:
+      ensure    => installed,
+      source    => $ssh_package_source_real,
+      adminfile => $ssh_package_adminfile_real,
+      before    => [
+        File['ssh_config'],
+        File['sshd_config'],
+      ],
+    }
   }
 
   file  { 'ssh_config' :
     ensure  => file,
     path    => $ssh_config_path,
     owner   => $ssh_config_owner,
-    group   => $ssh_config_group,
+    group   => $ssh_config_group_real,
     mode    => $ssh_config_mode,
     content => template($ssh_config_template),
-    require => Package[$packages_real],
   }
 
   file  { 'sshd_config' :
     ensure  => file,
     path    => $sshd_config_path,
     mode    => $sshd_config_mode_real,
     owner   => $sshd_config_owner,
-    group   => $sshd_config_group,
+    group   => $sshd_config_group_real,
     content => template($sshd_config_template),
-    require => Package[$packages_real],
   }
 
   if $sshd_config_banner != 'none' and $sshd_banner_content != undef {
-    file { 'sshd_banner' :
-      ensure  => file,
-      path    => $sshd_config_banner,
-      owner   => $sshd_banner_owner,
-      group   => $sshd_banner_group,
-      mode    => $sshd_banner_mode,
-      content => $sshd_banner_content,
-      require => Package[$packages_real],
+    if $packages_real == undef {
+      file { 'sshd_banner' :
+        ensure  => file,
+        path    => $sshd_config_banner,
+        owner   => $sshd_banner_owner,
+        group   => $sshd_banner_group,
+        mode    => $sshd_banner_mode,
+        content => $sshd_banner_content,
+        require => undef,
+      }
+    } else {
+      file { 'sshd_banner' :
```

Comment:
This I dont like because it violates DRY by having me repeat the resource. The other approach would be to use the inline `?` operator, though that is generally frowned up, though given this case, it may be the correct way to go.

The other approach would be to generate a string for the require, though using a template for some string manipulation in ruby sounds even uglier.